pkg/controllers/management/auth/globalroles/globalrolebinding_handler.go

```diff
@@ -205,6 +225,48 @@ func (grb *globalRoleBindingLifecycle) deleteAdminBinding(obj *v3.GlobalRoleBind
 	return nil
 }
 
+func (grb *globalRoleBindingLifecycle) reconcileSubject(binding *v3.GlobalRoleBinding, localConditions *[]metav1.Condition) (*v3.GlobalRoleBinding, error) {
+	condition := metav1.Condition{Type: subjectReconciled}
+
+	if binding.UserPrincipalName != "" && binding.UserName != "" {
```

Comment:
There is no need to reconcile the subject if the `GroupPrincipal` is present. A GRB with `GroupPrincipal` , but without a `UserName` or `UserPrincipal` is a valid GRB.
I think we should add `GroupPrincipal`  to this condition like we do for [crtbs](https://github.com/rancher/rancher/blob/main/pkg/controllers/management/auth/crtb_handler.go#L134) and [prtbs](https://github.com/rancher/rancher/blob/main/pkg/controllers/management/auth/prtb_handler.go#L99)